Improvements by vuidteam

laurieodgers opened the pull request:
Added:
- output of ip4 to command "iocage list" by default. If ip4 is not set, "-" will be output. 
- new command listby: this reuses the list function and provides a convenient sorted output. Each column can be sorted upon (including the new ip4).

Fixed:
- When run in background, iocage create would cause the background script to stop, and would not complete the creation of the jail. This is due to setfib waiting for data from stdin. Fix to this is to echo /dev/null to setfib.

skarekrow commented:
Thanks for this! A few nitpicks. We use all lowercase for our variable names, no need for `==`, we use just `=` now. Also I'd rather have this be called via `list` itself. So `iocage list -s uuid`, etc. 

Since I don't use fib's, I'll defer to @pannon on if that echo causes any harm.

laurieodgers commented:
@pannon do you have any comments on the setfib before I go ahead and fix up this pull request?

laurieodgers commented:
@skarekrow I have updated my changes as per your requests. Sorting is currently only done on iocage jails, with the command "iocage list -s <column>" where <column> is the name of the column to sort on.

I have removed the setfib change from this request and will resubmit it in another request once this one is complete.

skarekrow commented on lib/ioc-info:
```diff
@@ -130,51 +130,103 @@ __list_jails () {
+    {
+        # print out the ioc jails head
+        printf "%s^%s^%s^%s^%s^%s^%s\n" \
+        "JID" "UUID" "BOOT" "STATE" "TAG" "TYPE" "IP4" 
```
Indent 4 spaces please :)

skarekrow commented on lib/ioc-info:
```diff
@@ -130,51 +130,103 @@ __list_jails () {
+        ;;
+        ip4) 
+            _ioc_jailbody=$( echo "${_ioc_jailbody}" | awk -F "^" '{print $7 " % " $0}'\
+            | sort -t. -n -k1,1 -k2,2 -k3,3 -k4,4 | sed 's/[^%]*% //' )
```
Indent 4 spaces please :)

skarekrow commented:
Looks good! Can you do those 2 indentations and then squash your PR to one commit please?

Thanks!
